ginivision/src/main/java/net/gini/android/vision/analysis/AnalysisFragmentImpl.java

```diff
@@ -110,15 +133,111 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
         return view;
     }
 
+    private Handler mHandler = new Handler(Looper.getMainLooper());
+
     public void onDestroy() {
         mPhoto = null;
         stopScanAnimation();
     }
 
     public void onStart() {
+        showHints();
+    }
+
+    private void showHints() {
+
+        final int containerHeight = getContainerHeight();
+        mHints = generateRandomHintsList();
+
+        mRunnable = new Runnable() {
+            @Override
+            public void run() {
+                mHintAnimation = getSlideDownAnimation(containerHeight);
+                mHintAnimation.start();
+            }
+        };
+        mHandler.postDelayed(mRunnable, HINT_START_DELAY);
+    }
+
+    @NonNull
+    private ViewPropertyAnimatorCompat getSlideDownAnimation(final int containerHeight) {
+        return ViewCompat.animate(mHintContainer)
+                .translationY(containerHeight)
+                .setDuration(HINT_ANIMATION_DURATION)
+                .setListener(new ViewPropertyAnimatorListener() {
+                    @Override
+                    public void onAnimationStart(final View view) {
+                    }
+
+                    @Override
+                    public void onAnimationEnd(final View view) {
+                        setNextHint();
+                        mHintAnimation = getSlideUpAnimation();
+                        mHintAnimation.start();
+                    }
+
+                    @Override
+                    public void onAnimationCancel(final View view) {
+                    }
+                });
     }
 
-    public void onStop() {
+    @NonNull
+    private ViewPropertyAnimatorCompat getSlideUpAnimation() {
+        return ViewCompat.animate(mHintContainer)
+                .translationY(0)
+                .setDuration(HINT_ANIMATION_DURATION)
+                .setListener(new ViewPropertyAnimatorListener() {
+                    @Override
+                    public void onAnimationStart(final View view) {
+                    }
+
+                    @Override
+                    public void onAnimationEnd(final View view) {
+                        mHandler.postDelayed(mRunnable, HINT_CYCLE_INTERVAL);
+                    }
+
+                    @Override
+                    public void onAnimationCancel(final View view) {
+                    }
+                });
+    }
+
+    private int getContainerHeight() {
+        final Activity activity = mFragment.getActivity();
+        final Display display = activity.getWindowManager().getDefaultDisplay();
+        final Point size = new Point();
+        display.getSize(size);
```

Comment:
If the fragment is not shown full screen, we might not get the size we are looking for. What about getting the layout root and getting it's height? Would have to use the ViewTreeObserver to get the final height if there's no better way you know of...